explorer/data/prelude.carbon

```diff
@@ -416,6 +416,56 @@ external impl i32 as RightShiftWith(i32) where .Result == i32 {
   }
 }
 
+//-------------------------
+// Optional
+//-------------------------
+choice OptionalElement(T:! Type) {
+  None(),
+  Element(T)
+}
+
+class Optional(T:! Type){
+  fn CreateEmpty() -> Optional(T){
+    return { .element = OptionalElement(T).None() };
+  }
+  fn Create ( value: T ) -> Optional(T){
+    return { .element = OptionalElement(T).Element(value) };
+  }
+
+  fn has_value[me: Self] () -> bool{
+    var x: OptionalElement(T) = me.element;
+    match(x){
+      case OptionalElement(T).None() => { return false; }
+    }
+    return true;
+  }
+
+  fn get[me: Self] () -> T {
+    var y: T;
+    var x: OptionalElement(T) = me.element;
+    match(x){
+      case OptionalElement(T).Element( var x: T ) =>{
```

Comment:
```suggestion
      case OptionalElement(T).Element(x: T) =>{
```
I don't think we should need a `var` binding here, given that we don't modify `x`.

Reply:
var is removed!